library/src/scripts/components/mebox/pieces/VanillaHeaderNavItem.tsx

```diff
@@ -0,0 +1,56 @@
+/*
+ * @author Stéphane LaFlèche <stephane.l@vanillaforums.com>
+ * @copyright 2009-2018 Vanilla Forums Inc.
+ * @license GPL-2.0-only
+ */
+
+import * as React from "react";
+import classNames from "classnames";
+import { withRouter, RouteComponentProps } from "react-router-dom";
+import SmartLink from "@library/components/navigation/SmartLink";
+
+export interface IVanillaHeaderNavItemProps {
+    className?: string;
+    to: string;
+    name: string;
+    linkClassName?: string;
+    linkContentClassName?: string;
+}
+
+interface IProps extends IVanillaHeaderNavItemProps, RouteComponentProps<{}> {}
+
+/**
+ * Implements Navigation item component for header
+ */
+export class VanillaHeaderNavItem extends React.Component<IProps> {
```

Comment:
Can we get an issue for the spec on when these are supposed to marked as active? It's a bit of a mess in vanilla's current nav.

I'm thinking sub-path matching should work. Eg. If this link is to `/kb` and I'm on `/kb/articles/something` it should be active? Maybe we need to get more granular and offer a mode of what marks it as active.

Reply:
https://github.com/vanilla/knowledge/issues/329